Source/Core/VideoCommon/PostProcessing.h

```diff
@@ -4,28 +4,97 @@
 
 #pragma once
 
+#include <array>
 #include <map>
+#include <memory>
 #include <string>
+#include <unordered_map>
 #include <vector>
 
+#include "Common/Flag.h"
 #include "Common/StringUtil.h"
 #include "Common/Timer.h"
 
 #include "VideoCommon/VideoCommon.h"
 
+enum PostProcessingTrigger : u32
+{
+	POST_PROCESSING_TRIGGER_ON_SWAP,
+	POST_PROCESSING_TRIGGER_ON_PROJECTION,
+	POST_PROCESSING_TRIGGER_ON_EFB_COPY
+};
+
+enum PostProcessingOptionType : u32
+{
+	POST_PROCESSING_OPTION_TYPE_BOOL,
+	POST_PROCESSING_OPTION_TYPE_FLOAT,
+	POST_PROCESSING_OPTION_TYPE_INTEGER
+};
+
+enum PostProcessingInputType : u32
+{
+	POST_PROCESSING_INPUT_TYPE_IMAGE,                   // external image loaded from file
+	POST_PROCESSING_INPUT_TYPE_COLOR_BUFFER,            // colorbuffer at internal resolution
+	POST_PROCESSING_INPUT_TYPE_DEPTH_BUFFER,            // depthbuffer at internal resolution
+	POST_PROCESSING_INPUT_TYPE_PREVIOUS_PASS_OUTPUT,    // output of the previous pass
+	POST_PROCESSING_INPUT_TYPE_PASS_OUTPUT              // output of a previous pass
+};
+
+enum PostProcessingInputFilter : u32
+{
+	POST_PROCESSING_INPUT_FILTER_NEAREST,               // nearest/point sampling
+	POST_PROCESSING_INPUT_FILTER_LINEAR                 // linear sampling
+};
+
+enum PostProcessingAddressMode : u32
+{
+	POST_PROCESSING_ADDRESS_MODE_CLAMP,                 // clamp to edge
+	POST_PROCESSING_ADDRESS_MODE_WRAP,                  // wrap around at edge
+	POST_PROCESSING_ADDRESS_MODE_BORDER                 // fixed color (0) at edge
+};
+
+// Type describing a screen resolution as integer coordinates.
+// TODO: Move this to VideoCommon.h, or the PostProcessor class?
+struct TargetSize final
+{
+	TargetSize() : width(1), height(1) {}
+	TargetSize(int new_width, int new_height) : width(new_width), height(new_height) {}
+	TargetSize(const TargetSize &copy) : width(copy.width), height(copy.height) {}
+
+	void Set(int new_width, int new_height) { width = new_width; height = new_height; }
+
+	TargetSize& operator=(const TargetSize& rhs) { width = rhs.width; height = rhs.height; return *this; }
+
+	bool operator==(const TargetSize& rhs) const { return std::tie(width, height) == std::tie(rhs.width, rhs.height); }
+	bool operator!=(const TargetSize& rhs) const { return std::tie(width, height) != std::tie(rhs.width, rhs.height); }
+	bool operator<=(const TargetSize& rhs) const { return std::tie(width, height) <= std::tie(rhs.width, rhs.height); }
+	bool operator>=(const TargetSize& rhs) const { return std::tie(width, height) >= std::tie(rhs.width, rhs.height); }
+	bool operator<(const TargetSize& rhs) const { return std::tie(width, height) < std::tie(rhs.width, rhs.height); }
+	bool operator>(const TargetSize& rhs) const { return std::tie(width, height) > std::tie(rhs.width, rhs.height); }
+
+	int width;
+	int height;
```

Comment:
These can be directly assigned
```cpp
int width = 1;
int height = 1;
```

and have the default constructor turned into

```cpp
constexpr TargetSize() = default;
```